src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,30 +36,66 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        addressFields = trimmedAddress.split(FIELD_SEPARATOR);
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(addressFields)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        
+        setFields(addressFields);
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String[] test) {
+        return test.length == TOTAL_FIELDS && PostalCode.isValidPostalCode(test[POSTAL_CODE_INDEX]);
```

Comment:
this is a better validation than the default. however, it is not very robust (think why?).
you can try to make the validation more robust if you wish to (think how!)